src/extensions/scratch3_text2speech/index.js

```diff
@@ -366,6 +449,15 @@ class Scratch3Text2SpeechBlocks {
         // Only set the language if it is in the list.
         if (this.isSupportedLanguage(languageCode)) {
             stage.textToSpeechLanguage = languageCode;
+        } else if (languageCode === 'ja-Hira') {
+            // Japanese (Hiragana) is equal to Japanese here
+            stage.textToSpeechLanguage = 'ja';
```

Comment:
A lot of this seems to be specifically aimed at Hiragana. Are there any other languages with writing systems that might have the same problem as Hiragana does for Japanese?

Reply:
Maybe it would be a good idea to change `isSupportedLanguage` such that it is aware of the prefixes, and performs the conversion automatically for every language of the form `xx-yy`. Alternatively, you could add an easy-to-extend array/object of such languages and search through it (for now, it would only have one entry).

Perhaps, in the future, the TTS extension might support more languages.

Reply:
I hope it does